### PR TITLE
[UWP]: Don't use AdaptiveImage to render BackgroundImage

### LINKED ISSUE #2712: [UWP] Don't re-use custom AdaptiveImage renderer for background images
Using the AdaptiveImage renderer to render background images causes strange things. As a host, if I override rendering of AdaptiveImage, I also override rendering of background images, which probably wasn't my intention.

image

At the working group meeting today (with Matt, David, Andrew, Bill, Alberto, Shalini, Becky, and Paul), consensus was that we shouldn't be doing this.

When a need arises for overriding background image rendering, we'll solve that with a separate API

### PR DESCRIPTION
**Pull request by beervoley**
# Related Issue
Closes #2712.

# Description
We opt out of using `AdaptiveImageRenderer` in the case of rendering a background image. 
This will allow to override `AdaptiveImageRenderer` without overriding background image rendering.

# Sample Card
Old implementation on the left, new implementation on the right:

image

# How Verified
Manually.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6838)

### CODE REVIEW

File: source/uwp/Renderer/lib/XamlHelpers.cpp [hunk 1]
```diff
@@ -359,44 +359,48 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
...
 
-        ComPtr<IAdaptiveCardElement> adaptiveCardElement;
-        THROW_IF_FAILED(adaptiveImage.As(&adaptiveCardElement));
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        THROW_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
 
-        ComPtr<IAdaptiveElementRendererRegistration> elementRenderers;
-        THROW_IF_FAILED(renderContext->get_ElementRenderers(&elementRenderers));
+        ComPtr<IUriRuntimeClass> imageUrl;
+        GetUrlFromString(hostConfig.Get(), url, imageUrl.GetAddressOf());
```
Comment (beervoley) [inline source/uwp/Renderer/lib/XamlHelpers.cpp:373]: @paulcam206 should I do this in **applyBackgroundToRoot** and pass an actual working url here?

File: source/uwp/Renderer/lib/XamlHelpers.cpp [hunk 2]
```diff
@@ -359,44 +359,48 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
...
         ComPtr<IImage> xamlImage;
-        THROW_IF_FAILED(background.As(&xamlImage));
+        CreateBackgroundImage(renderContext, url.Get(), &xamlImage);
+        if (!xamlImage)
+        {
+            return;
+        }
 
-        ABI::AdaptiveCards::ObjectModel::Uwp::BackgroundImageFillMode fillMode;
-        THROW_IF_FAILED(backgroundImage->get_FillMode(&fillMode));
```
Comment (paulcam206) [inline source/uwp/Renderer/lib/XamlHelpers.cpp:399]: this is just simple dead code, yes? have you verified that [`fillMode`](https://adaptivecards.io/explorer/BackgroundImage.html) still works?

**Review by paulcam206 — APPROVED** [review]

File: source/uwp/Renderer/lib/XamlHelpers.cpp [hunk 3]
```diff
@@ -359,44 +359,48 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
         return S_OK;
     }
 
-    void ApplyBackgroundToRoot(_In_ IPanel* rootPanel,
-                               _In_ IAdaptiveBackgroundImage* backgroundImage,
-                               _In_ IAdaptiveRenderContext* renderContext,
-                               _In_ IAdaptiveRenderArgs* renderArgs)
+    void CreateBackgroundImage(_In_ IAdaptiveRenderContext* renderContext, _In_ HSTRING url, _Outptr_ IImage** backgroundImage)
```
Comment (licanhua) [inline source/uwp/Renderer/lib/XamlHelpers.cpp:362]: nit: HSTRING const& url

**Review by licanhua — APPROVED** [review]

File: source/uwp/Renderer/lib/XamlHelpers.cpp (hunk 3 above)
Comment (beervoley) [reply source/uwp/Renderer/lib/XamlHelpers.cpp:362]: @licanhua
image
there's no need for const& :)

File: source/uwp/Renderer/lib/XamlHelpers.cpp (hunk 2 above)
Comment (beervoley) [reply source/uwp/Renderer/lib/XamlHelpers.cpp:399]: **fillMode** is being applied in **TileControl**, checking now..

**Pull request merged**